knowagewhatifengine/src/main/java/it/eng/spagobi/engines/whatif/calculatedmember/MDXFormulaHandler.java

```diff
@@ -196,10 +205,34 @@ private static void injectTimeDimensionInBody() {
 
 class Box<T> {
 
+	private static final Logger LOGGER = Logger.getLogger(Box.class);
+
 	@SuppressWarnings("unchecked")
 	public T unmarshalFile(File file, Class<T> clazz) throws JAXBException {
 
 		JAXBContext jc = JAXBContext.newInstance(clazz);
+		XMLInputFactory xif = XMLInputFactory.newFactory();
+		xif.setProperty(XMLInputFactory.SUPPORT_DTD, false);
+		xif.setProperty(XMLInputFactory.IS_REPLACING_ENTITY_REFERENCES, false);
+		xif.setProperty(XMLInputFactory.IS_SUPPORTING_EXTERNAL_ENTITIES, false);
+
+		FileInputStream inputStream = null;
+		try {
+			inputStream = new FileInputStream(file);
+			InputStreamReader reader = new InputStreamReader(inputStream, StandardCharsets.UTF_8);
+			XMLStreamReader xsr = xif.createXMLStreamReader(new StreamSource(reader));
+
+			return (T) jc.createUnmarshaller().unmarshal(xsr);
+		} catch (FileNotFoundException | XMLStreamException e) {
+			LOGGER.error("Error loading XML document: " + e.getMessage(), e);
+		} finally {
+			try {
+				inputStream.close();
+			} catch (IOException e) {
+				LOGGER.error("Error loading XML document: " + e.getMessage(), e);
+			}
+		}
+
 		Unmarshaller unmarshaller = jc.createUnmarshaller();
 		return (T) unmarshaller.unmarshal(file);
```

Comment:
Are those lines still useful?

Reply:
I think so, I think it is used for creating Java object from parsed XML.

https://www.tabnine.com/code/java/methods/javax.xml.bind.Unmarshaller/unmarshal

I'm working on other changes.

Reply:
Possibly fixed (with issue below).